TAScheduler/AcceptanceTests/test_edit_section.py

```diff
@@ -0,0 +1,102 @@
+        # Creating 3 Courses: hardcoded "numSec","modality","credits"
+        for i in [1, 2, 3]:
+            testCourse = Course(course_id=i, semester="semester" + str(i), name="name" + str(i), num_of_sections=2,
+                                modality="Remote", credts=3, )
+            testCourse.save()
+            self.courseList = self.courseList.append(testCourse)
+        # Creating 3 Sections: hardcoded "meetingTime"
+        for i in [1, 2, 3]:
+            testSection = Section(section_id=i, course=self.courseList[i - 1], location="location" + str(i),
+                                  meeting_time=datetime(2023, 1, 1, 1, 1, 1))
+            testSection.save()
+            self.secList = self.secList.append(testSection)
+
+    # 1] Database reflects our input
+    def test_DatabaseEditChange(self):
+        secChanges = {"location": "Naboo"}
+        resp = self.client.post("home/managesection/edit", data={"section": self.secList[0], "edit": secChanges})
+        dbEditedSect = Section.objects.filter(section_id=self.secList[0].section_id)
+        self.assertEquals(dbEditedSect["location"], "Naboo")
+
+    # 2] Confirmation message
+    def test_ConfirmationMessage(self):
+        secChanges = {"location": "Naboo"}
+        resp = self.client.post("home/managesection/edit", data={"section": self.secList[0], "edit": secChanges})
+        self.assertEquals(resp.context["message"], "SUCCESSFULLY CHANGED INFORMATION")
+
+
+# AC2] - Unsuccessful (invalid changes)
+class UnSuccessEdit(TestCase):
+    def setUp(self):
+        self.client = Client()
+        self.courseList = list()  # holding list of test courses so easier to call later
+        self.secList = list()
+        # Creating 3 Courses: hardcoded "numSec","modality","credits"
+        for i in [1, 2, 3]:
+            testCourse = Course(course_id=i, semester="semester" + str(i), name="name" + str(i), num_of_sections=2,
+                                modality="Remote", credts=3, )
+            testCourse.save()
+            self.courseList = self.courseList.append(testCourse)
+        # Creating 3 Sections: hardcoded "meetingTime"
+        for i in [1, 2, 3]:
+            testSection = Section(section_id=i, course=self.courseList[i - 1], location="location" + str(i),
+                                  meeting_time=datetime(2023, 1, 1, 1, 1, 1))
+            testSection.save()
+            self.secList = self.secList.append(testSection)
+
+    def test_MissingFields(self):
+        secChanges = Section(section_id="", course="",location="",meeting_time="")
+        resp = self.client.post("home/managesection/edit", data={"section": self.secList[0], "edit": secChanges})
+        self.assertEquals(resp.context["message"], "MISSING SECTION FIELDS")
+
+    #add these if we're going to allow editing of lectures/labs.
+    #def test_NonExistantUserAsgmt(self):
+    #def test_NonExistantCourse(self):
+
+#
+#
+#
+#
+
+# AC3] - Discard
+class discardEdit(TestCase):
```

Comment:
If we are discarding a change we don't submit the form so no post method gets called. Therefore nothing changes. I imagine the discard button as a link back to the edit section selection page

Reply:
I thought since "discard" is a button press, there's a submission. I asked chatGPT, and it said in the submission maybe there could be 2 submit buttons (save,discard), if we click the discard one,
 it's saying all we need to do is "if 'save_changes' in request.POST" 

"When you press the "Save Changes" button, the form is submitted with the name-value pair corresponding to that button (save_changes=Save Changes). If you press the "Discard Changes" button, the form is submitted with the name-value pair (discard_changes=Discard Changes"

That's also it's response^ - What do you think?